src/main/ruby/core/truffle/cext.rb

```diff
@@ -102,13 +102,18 @@ def [](index)
     end
   end
 
-  class RStringPtr
+  class AbstractRStringPtr
     attr_reader :string
 
     def initialize(string)
       @string = string
     end
 
+    alias_method :to_str, :string
+    alias_method :to_s, :string
```

Comment:
Do to_str/to_s need to be defiend for RStringPtrEnd ?
i.e. I think rb_str_new_cstr(RSTRING_END(str)) should return an empty String, not the full str.